### PR TITLE
Replace deprecated library PhpExcel by PhpSpreadsheet

### LINKED ISSUE #10962: Replace deprecated library PhpExcel by PhpSpreadsheet
**Is your feature request related to a problem? Please describe.**

Library `phpoffice/phpexcel` is used in PrestaShop Core: https://github.com/PrestaShop/PrestaShop/blob/develop/composer.json#L38

According to its [github repository README](https://github.com/PHPOffice/PHPExcel/blob/1.8/README.md) the library is deprecated since 2015. Using it now in 2018 is not efficient and could even introduce security flaws.

**Describe the solution you'd like**
The github repository README indicates that https://github.com/PHPOffice/PhpSpreadsheet should be used instead. So solution is to replace `PhpExcel` by `PhpSpreadsheet`.

### PR DESCRIPTION
**Pull request by MartinFinkenflugel**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | Replaced all usage of PHPExcel by PHPSpreadsheet
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #10962 
| How to test?  | Import an XLS or XLSX file from the admin view

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11246)

### CODE REVIEW

File: controllers/admin/AdminImportController.php [hunk 1]
```diff
@@ -4280,11 +4280,11 @@ protected function excelToCsvFile($filename)
             }
 
             if (!is_file($dest_file)) {
-                $reader_excel = PHPExcel_IOFactory::createReaderForFile($csv_folder . $filename);
+                $reader_excel = \PhpOffice\PhpSpreadsheet\IOFactory::createReaderForFile($csv_folder . $filename);
```
Comment (Quetzacoalt91) [inline controllers/admin/AdminImportController.php]: Can you please declare the namespace at the top of the file?

```php
use PhpOffice\PhpSpreadsheet\IOFactory;
```

Once done, you can replace this `\PhpOffice\PhpSpreadsheet\IOFactory` with `IOFactory`.

Comment (Quetzacoalt91) [reply controllers/admin/AdminImportController.php]:
```suggestion
                $reader_excel = IOFactory::createReaderForFile($csv_folder . $filename);
```

File: controllers/admin/AdminImportController.php [hunk 2]
```diff
@@ -4280,11 +4280,11 @@ protected function excelToCsvFile($filename)
...
                 $reader_excel->setReadDataOnly(true);
                 $excel_file = $reader_excel->load($csv_folder . $filename);
 
-                $csv_writer = PHPExcel_IOFactory::createWriter($excel_file, 'CSV');
+                $csv_writer = \PhpOffice\PhpSpreadsheet\IOFactory::createWriter($excel_file, 'Csv');
```
Comment (Quetzacoalt91) [inline controllers/admin/AdminImportController.php]:
```suggestion
                $csv_writer = IOFactory::createWriter($excel_file, 'Csv');
```

File: src/Adapter/Converter/ExcelToCsvFileConverter.php [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Converter;
 
-use PHPExcel_IOFactory;
+use \PhpOffice\PhpSpreadsheet\IOFactory;
```
Comment (Quetzacoalt91) [inline src/Adapter/Converter/ExcelToCsvFileConverter.php]:
```suggestion
use PhpOffice\PhpSpreadsheet\IOFactory;
```

File: src/Adapter/Converter/ExcelToCsvFileConverter.php [hunk 4]
```diff
@@ -81,10 +81,10 @@ public function convert(SplFileInfo $sourceFile)
         $destinationFilePath = $this->excelDirectory . $destinationFilename;
 
         if (!$this->filesystem->exists($destinationFilePath)) {
-            $excelReader = PHPExcel_IOFactory::createReaderForFile($sourceFile->getFilename());
+            $excelReader = \PhpOffice\PhpSpreadsheet\IOFactory::createReaderForFile($sourceFile->getFilename());
```
Comment (Quetzacoalt91) [inline src/Adapter/Converter/ExcelToCsvFileConverter.php]:
```suggestion
            $excelReader = IOFactory::createReaderForFile($sourceFile->getFilename());
```

File: src/Adapter/Converter/ExcelToCsvFileConverter.php [hunk 5]
```diff
@@ -81,10 +81,10 @@ public function convert(SplFileInfo $sourceFile)
...
             $excelReader->setReadDataOnly(true);
             $excelFile = $excelReader->load($sourceFile->getFilename() . $destinationFilename);
-            $csvWriter = PHPExcel_IOFactory::createWriter($excelFile, 'CSV');
+            $csvWriter = \PhpOffice\PhpSpreadsheet\IOFactory::createWriter($excelFile, 'Csv');
```
Comment (Quetzacoalt91) [inline src/Adapter/Converter/ExcelToCsvFileConverter.php]:
```suggestion
            $csvWriter = IOFactory::createWriter($excelFile, 'Csv');
```

**Review by Quetzacoalt91 — CHANGES_REQUESTED** [review]
Thank you for replacing the lib. Here a few changes to bring regarding the namespace declaration.

File: controllers/admin/AdminImportController.php (hunk 1 above)
Comment (MartinFinkenflugel) [reply controllers/admin/AdminImportController.php]: Thanks for the suggestion, that's a neater way of doing it.

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks a lot @MartinFinkenflugel

**Comment by matks** [comment]
@MartinFinkenflugel Just to let you know that this PR was finally merged through https://github.com/PrestaShop/PrestaShop/pull/14316

**Pull request merged**